File: test/tree/design-in-memory-file-system.test.ts
import { FileSystem } from '../../src/tree/design-in-memory-file-system';

describe('design in memory file system', () => {
  test('design in memory file system - test case 1', async () => {
    const fs = new FileSystem();

    expect(fs.ls('/')).toStrictEqual([]);

    fs.mkdir('/a/b/c');
    fs.addContentToFile('/a/b/c/d', 'hello');

    expect(fs.ls('/')).toStrictEqual(['a']);
    expect(fs.readContentFromFile('/a/b/c/d')).toBe('hello');
  });
});

File: test/leetcode/tree/design_in_memory_file_system_test.py
from leetcode.tree.design_in_memory_file_system import FileSystem


def test_case_1():
    fs = FileSystem()

    assert fs.ls("/") == []

    fs.mkdir("/a/b/c")
    fs.addContentToFile("/a/b/c/d", "hello")

    assert fs.ls("/") == ["a"]
    assert fs.readContentFromFile("/a/b/c/d") == "hello"
